### ISSUE TITLE
terminate called after throwing an instance of 'dmlc::Error'

### ISSUE THREAD

**Issue by alokgogate** [dmlc/xgboost#4623]
Operating system: Centos 7
xgboost version: 0.90
xgboost installed from pypip repository: `pip install xgboost`
I get the error for the following code after trying to fit a xgboost classifier on a very large dataset (~100 million rows):

code:
```
from xgboost import XGBClassifier
train = pd.read_parquet("/data/large_train.parquet")
y_train = train['Target']
X_train = train.drop(['Target'],axis=1)

model = XGBClassifier(nthreads=-1,random_state=0,n_estimators=400)

mode.fit(X_train,y_train)
```
error:
> terminate called after throwing an instance of 'dmlc::Error'
>   what():  [10:34:31] src/tree/updater_histmaker.cc:311: fv=inf, hist.last=inf
> Stack trace:
>   [bt] (0) /root/anaconda3/lib/python3.6/site-packages/xgboost/./lib/libxgboost.so(dmlc::LogMessageFatal::~LogMessageFatal()+0x22) [0x7f275d854f32]
>   [bt] (1) /root/anaconda3/lib/python3.6/site-packages/xgboost/./lib/libxgboost.so(xgboost::tree::CQHistMaker::HistEntry::Add(float, xgboost::detail::GradientPairInternal<float>)+0x3b2) [0x7f275d9a4302]
>   [bt] (2) /root/anaconda3/lib/python3.6/site-packages/xgboost/./lib/libxgboost.so(+0x1eb9e9) [0x7f275d99e9e9]
>   [bt] (3) /root/anaconda3/lib/python3.6/site-packages/xgboost/./lib/libxgboost.so(+0x1ee4e5) [0x7f275d9a14e5]
>   [bt] (4) /root/anaconda3/bin/../lib/libgomp.so.1(+0xfb05) [0x7f275d59fb05]
>   [bt] (5) /lib64/libpthread.so.0(+0x7dd5) [0x7f2818a6fdd5]
>   [bt] (6) /lib64/libc.so.6(clone+0x6d) [0x7f2818798ead]

**Comment by trivialfis** [comment]
@alokgogate Could you please post some sample data so that I can reproduce?

**Comment by alokgogate** [comment]
@trivialfis unfortunately I wont be able to post sample data as the data is confidential. However I did manage to run it successfully on a small subset (~5 million rows) of the large dataset.

**Comment by techdozor** [comment]
Very similar issue and somehow once I hop over 5m rows, it just dies...

I am on ubuntu

`Fitting 10 folds for each of 1 candidates, totalling 10 fits
[Parallel(n_jobs=1)]: Using backend SequentialBackend with 1 concurrent workers.
[CV] max_depth=3, learning_rate=0.5, n_estimators=50 .................
terminate called after throwing an instance of 'dmlc::Error'
  what():  [23:14:02] /workspace/src/tree/updater_histmaker.cc:313: fv=inf, hist.last=inf
Stack trace:
  [bt] (0) /home/ubuntu/.local/lib/python3.5/site-packages/xgboost/./lib/libxgboost.so(dmlc::LogMessageFatal::~LogMessageFatal()+0x54) [0x7f54715d8614]
  [bt] (1) /home/ubuntu/.local/lib/python3.5/site-packages/xgboost/./lib/libxgboost.so(xgboost::tree::CQHistMaker::HistEntry::Add(float, xgboost::detail::GradientPairInternal<float>)+0x3cf) [0x7f547177077f]
  [bt] (2) /home/ubuntu/.local/lib/python3.5/site-packages/xgboost/./lib/libxgboost.so(xgboost::tree::CQHistMaker::UpdateHistCol(std::vector<xgboost::detail::GradientPairInternal<float>, std::allocator<xgboost::detail::GradientPairInternal<float> > > const&, xgboost::common::Span<xgboost::Entry const, 18446744073709551615ul> const&, xgboost::MetaInfo const&, xgboost::RegTree const&, std::vector<unsigned int, std::allocator<unsigned int> > const&, unsigned int, std::vector<xgboost::tree::CQHistMaker::HistEntry, std::allocator<xgboost::tree::CQHistMaker::HistEntry> >*)+0x55f) [0x7f547177675f]
  [bt] (3) /home/ubuntu/.local/lib/python3.5/site-packages/xgboost/./lib/libxgboost.so(+0x2e748b) [0x7f547177748b]
  [bt] (4) /usr/lib/x86_64-linux-gnu/libgomp.so.1(GOMP_parallel+0x3f) [0x7f5471279cbf]
  [bt] (5) /home/ubuntu/.local/lib/python3.5/site-packages/xgboost/./lib/libxgboost.so(xgboost::tree::GlobalProposalHistMaker::CreateHist(std::vector<xgboost::detail::GradientPairInternal<float>, std::allocator<xgboost::detail::GradientPairInternal<float> > > const&, xgboost::DMatrix*, std::vector<unsigned int, std::allocator<unsigned int> > const&, xgboost::RegTree const&)+0x942) [0x7f5471787082]
  [bt] (6) /home/ubuntu/.local/lib/python3.5/site-packages/xgboost/./lib/libxgboost.so(xgboost::tree::HistMaker::UpdateTree(std::vector<xgboost::detail::GradientPairInternal<float>, std::allocator<xgboost::detail::GradientPairInternal<float> > > const&, xgboost::DMatrix*, xgboost::RegTree*)+0x464) [0x7f547177ed74]
  [bt] (7) /home/ubuntu/.local/lib/python3.5/site-packages/xgboost/./lib/libxgboost.so(xgboost::tree::HistMaker::Update(xgboost::HostDeviceVector<xgboost::detail::GradientPairInternal<float> >*, xgboost::DMatrix*, std::vector<xgboost::RegTree*, std::allocator<xgboost::RegTree*> > const&)+0xb8) [0x7f547176e3e8]
  [bt] (8) /home/ubuntu/.local/lib/python3.5/site-packages/xgboost/./lib/libxgboost.so(xgboost::gbm::GBTree::BoostNewTrees(xgboost::HostDeviceVector<xgboost::detail::GradientPairInternal<float> >*, xgboost::DMatrix*, int, std::vector<std::unique_ptr<xgboost::RegTree, std::default_delete<xgboost::RegTree> >, std::allocator<std::unique_ptr<xgboost::RegTree, std::default_delete<xgboost::RegTree> > > >*)+0xed0) [0x7f5471687980]`

**Comment by trivialfis** [comment]
Could you please post some sample data for reproducing?  I can't guess what's happening

**Comment by techdozor** [comment]
-[ RECORD 1 ]-+-------------------------------------
time          | 2020-04-19 03:30:00
key           | 905-DSE-09-OKDJB122
numer    | 156
done      | f
key_converted | 23498345981739795743954395873498543985

**Comment by techdozor** [comment]
so if you take over 5m records like this it dies.

**Comment by techdozor** [comment]
code sample
`seed = 1337
np.random.seed(seed)
cv = StratifiedKFold(n_splits=10, shuffle=True, random_state=seed)
cv.get_n_splits(predictArray_train)

max_depth = [3]
learning_rate = [0.5]
n_estimators = [50]

params_grid = {
        'max_depth': max_depth,
            'n_estimators': n_estimators,
                'learning_rate': learning_rate
}

params_fixed = {'nthread':10,
                'objective':'binary:logistic',
                                'silent': 1,
}

xgb_model = xgb.XGBClassifier(**params_fixed, seed=seed)

clf = GridSearchCV(estimator=xgb_model, param_grid=params_grid, cv=cv, scoring='accuracy', verbose=2)
predModelARCV = clf.fit(featureArray_train, predictArray_train, eval_metric='rmse')`

**Comment by trivialfis** [comment]
```
-[ RECORD 1 ]-+-------------------------------------
time | 2020-04-19 03:30:00
key | 905-DSE-09-OKDJB122
number | 156
done | f
key_converted | 23498345981739795743954395873498543985
```

I don't understand what's this?

**Comment by techdozor** [comment]
This is a representative of a record. There are millions of those records and this is record #1 with the following columns:
time, key, number, done (true/false), key_converted.
My code sample above shows what I have done, but basically the goal of classification is to predict "done" (true/false) based on the data features of key_converted and a number.
I would not attempt to understand the logic behind this use case, because there is one, trust me. However, what's important to note that if you take let's say 7m (vs 5m) of the above records you reproduce the issue.
Make sense?

**Comment by techdozor** [comment]
So, can xgboost support more then 3m records? I would hope so, where is the issue, any clues?

**Comment by alokgogate** [comment]
@techdozor, i was able to run XGboost with ~8M rows now after upgrading the version of XGB to 1.0.2

**Comment by trivialfis** [comment]
Will look into this later.  But for now if the issue occurs, please try hist or gpu hist tree method.
